beacon-chain/core/light-client/store.go

```diff
@@ -200,3 +319,73 @@ func (s *Store) LastOptimisticUpdate() interfaces.LightClientOptimisticUpdate {
 	defer s.mu.RUnlock()
 	return s.lastOptimisticUpdate
 }
+
+func (s *Store) MigrateToCold(ctx context.Context, finalizedRoot [32]byte) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// If there cache is empty (some problem in processing data), we can update the tail and skip migration.
+	// This is a safety check and should not happen in normal operation.
+	if len(s.cache.items) == 0 {
+		log.Debug("Non-finality cache is empty. Skipping migration.")
+		return nil
+	}
+
+	blk, err := s.beaconDB.Block(ctx, finalizedRoot)
+	if err != nil {
+		return errors.Wrapf(err, "failed to fetch block for finalized finalizedRoot %x", finalizedRoot)
+	}
+	if blk == nil {
+		return errors.Errorf("nil block for finalized finalizedRoot %x", finalizedRoot)
+	}
+	finalizedSlot := blk.Block().Slot()
+	headRoot := blk.Block().ParentRoot()
+
+	var headItem *cacheItem
+	var ok bool
+
+	headItem, ok = s.cache.items[headRoot]
+	if !ok {
+		log.Debugf("Head root %x not found in light client cache. Cleaning the broken part of the cache.", headRoot)
```

Comment:
```suggestion
		log.Debugf("Finalized block's parent root %#x not found in light client cache. Cleaning the broken part of the cache.", headRoot)
```
I think it's fine to use `head` in variable names even if they don't mean the actual head of the chain, but using it in logs will be confusing. Everyone will assume you mean the head of the chain